include/EASTL/internal/functional_base.h

```diff
@@ -232,7 +232,7 @@ namespace eastl
 		T& get() const EA_NOEXCEPT;
 
 		template <typename... ArgTypes>
-		typename eastl::result_of<T&(ArgTypes&&...)>::type operator() (ArgTypes&&...) const;
+		typename eastl::invoke_result<T&, ArgTypes&&...>::type operator() (ArgTypes&&...) const;
```

Comment:
I'm having trouble coming up with a case where it makes a difference, perhaps it doesn't, but I don't think the `&&` in `<T&, ArgTypes&&...>` is necessary. If we can get away with a bit less noise I recommend removing it here and below where it's implemented.
```suggestion
		typename eastl::invoke_result<T&, ArgTypes...>::type operator() (ArgTypes&&...) const;
```